Tools/setup/ubuntu.sh

```diff
@@ -104,56 +104,63 @@ sudo DEBIAN_FRONTEND=noninteractive apt-get -y --quiet --no-install-recommends i
 
 # Python3 dependencies
 echo
-echo "Installing PX4 Python3 dependencies"
-if [ -n "$VIRTUAL_ENV" ]; then
-	# virtual environments don't allow --user option
-	python -m pip install -r ${DIR}/requirements.txt
+echo $VERBOSE_BAR
+echo "🍻 Installing PX4 Python3 dependencies"
+echo $VERBOSE_BAR
+echo
+
+PYTHON_VERSION=$(python3 --version 2>&1 | awk '{print $2}')
+REQUIRED_VERSION="3.11"
+if [[ "$(printf '%s\n' "$REQUIRED_VERSION" "$PYTHON_VERSION" | sort -V | head -n1)" == "$REQUIRED_VERSION" ]]; then
+	python3 -m pip install --break-system-packages -r ${DIR}/requirements.txt
 else
-	# older versions of Ubuntu require --user option
-	python3 -m pip install --user -r ${DIR}/requirements.txt
+	if [ -n "$VIRTUAL_ENV" ]; then
+		# virtual environments don't allow --user option
+		python -m pip install -r ${DIR}/requirements.txt
+	else
+		python3 -m pip install --user -r ${DIR}/requirements.txt
+	fi
 fi
 
 # NuttX toolchain (arm-none-eabi-gcc)
 if [[ $INSTALL_NUTTX == "true" ]]; then
 
 	echo
-	echo "Installing NuttX dependencies"
+	echo $VERBOSE_BAR
+	echo "🍻 Installing NuttX dependencies"
+	echo $VERBOSE_BAR
+	echo
 
 	sudo DEBIAN_FRONTEND=noninteractive apt-get -y --quiet --no-install-recommends install \
 		automake \
 		binutils-dev \
 		bison \
-		build-essential \
 		flex \
-		g++-multilib \
-		gcc-multilib \
-		gdb-multiarch \
 		genromfs \
 		gettext \
 		gperf \
+		kconfig-frontends \
 		libelf-dev \
 		libexpat-dev \
 		libgmp-dev \
 		libisl-dev \
 		libmpc-dev \
 		libmpfr-dev \
-		libncurses5 \
-		libncurses5-dev \
-		libncursesw5-dev \
+		libncurses6 \
+		libncurses-dev \
+		libncursesw6 \
 		libtool \
 		pkg-config \
 		screen \
 		texinfo \
 		u-boot-tools \
 		util-linux \
 		vim-common \
+		g++-arm-linux-gnueabihf \
+		gcc-arm-linux-gnueabihf \
+		g++-aarch64-linux-gnu \
```

Comment:
probably shouldn't be default

```suggestion
		g++-aarch64-linux-gnu \
```